app/Services/UserService.php

```diff
@@ -0,0 +1,106 @@
+<?php
+namespace App\Services;
+
+use App\Dao\UserDao;
+use App\Models\User;
+use Illuminate\Http\Request;
+use App\Contracts\Dao\UserDaoInterface;
+use App\Contracts\Services\UserServiceInterface;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\Hash;
+
+class UserService implements UserServiceInterface
+{
+    protected $userDao;
+
+    public function __construct(UserDaoInterface $userDaoInterface)
+    {
+        $this->userDao = $userDaoInterface;
+    }
+
+    /**
+     * insert user
+     *
+     * @param Request $request
+     * @return null
+     */
+    public function insert(Request $request): null
+    {
+        $encrypted = Hash::make($request->password);
+
+        // $path = $request->file('img')->store('pfp');
+
+        $filename = $request->img->getClientOriginalName();
```

Comment:
Line numbers 33 ~ 43 should be in service layer. 
And may be you don't need to return for this function. UserDao->insert() doesn't return anything.